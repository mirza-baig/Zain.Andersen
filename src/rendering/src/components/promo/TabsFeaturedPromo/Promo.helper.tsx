// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
// Components
import { Headline } from 'src/helpers/Headline';
import { Subheadline } from 'src/helpers/Subheadline';
import { BodyCopy } from 'src/helpers/BodyCopy';
import { ImagePrimary } from 'src/helpers/Media';
import SingleButton from 'src/helpers/SingleButton/SingleButton';
import classNames from 'classnames';
import { getEnum } from 'lib/utils';
import { Eyebrow } from 'src/helpers/Eyebrow';
import { Caption } from 'src/helpers/Caption';

export type PromoProps =
  Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsFeaturedPromo.TabContent & {
    classes: { [key: string]: string };
  };

const Promo = (props: PromoProps) => {
  const { fields, classes } = props;

  const { themeName } = useTheme();

  if (!fields) {
    return null;
  }

  return (
    <>
      <div
        className={classNames(
          themeName === 'rba' && getEnum(fields.backgroundColor) === 'black'
            ? 'theme-black'
            : 'theme-gray',
          classes.promoContainer
        )}
      >
        <div className={classes.promoImageWrapper}>
          <ImagePrimary hideCaption={true} {...props} />
          {themeName === 'rba' && fields.primaryImageCaption.value && (
            <div className={classes.promoImageCaptionWrapper}>
              <div className={classes.promoImageCaption}>
                <Caption
                  italic={false}
                  isImageCaption={false}
                  caption={fields.primaryImageCaption}
                />
              </div>
            </div>
          )}
        </div>
        <div className={classes.promoContentWrapper}>
          {themeName === 'rba' && <Eyebrow classes={classes.promoEyebrow} fields={fields} />}
          <Headline classes={classes.promoHeadline} fields={fields} />
          <Subheadline classes={classes.promoSubheadline} fields={fields} />
          <BodyCopy classes={classes.promoBody} fields={fields} />
          <SingleButton fields={fields} />
        </div>
      </div>
    </>
  );
};

export default Promo;
