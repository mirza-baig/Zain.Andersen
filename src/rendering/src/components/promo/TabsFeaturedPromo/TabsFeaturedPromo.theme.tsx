// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const TabsFeaturedPromoTheme: ThemeFile = {
  aw: {
    classes: {
      headline: 'text-s ml:text-m font-bold',
      tabsClasses: {
        tabWrapper: 'flex flex-col ml:flex-row min-h-[542px]',
        tabHeadlineWrapper: 'flex items-center border-t pt-s mb-xxs uppercase',
        tabHeadlineIcon: 'mr-xxxs',
        tabHeadline: 'text-xs font-heavy',
        tabTitlesContainer: 'relative flex order-3 ml:order-none pr-s pl-s',
        leftArrowIcon:
          'absolute right-full top-1/2 -translate-y-1/2 cursor-pointer bg-white before:absolute before:left-full before:inline-block before:h-[30px] before:w-[30px] before:bg-gradient-to-r before:from-white before:content-[""] ml:hidden',
        rightArrowIcon:
          'absolute left-full top-1/2 -translate-y-1/2 cursor-pointer bg-white before:absolute before:right-full before:inline-block before:h-[30px] before:w-[30px] before:bg-gradient-to-l before:from-white before:content-[""] ml:hidden',
        tabTitlesWrapper: 'order-3 flex ml:inline-block overflow-auto snap-x ml:snap-none',
        tabTitlesList: 'w-full ml:w-1/4 ml:pr-s flex flex-col ml:inline-block',
        tabTitle:
          'text-body ml:text-xs text-dark-gray cursor-pointer mx-xxs ml:mx-0 py-xxxs ml:px-xxs ml:border-l-2 my-s snap-start',
        activeTabTitle: '!text-secondary font-heavy border-b-2 ml:border-b-0 border-primary',
        tabPanel: 'w-full ml:w-3/4',
        toggleContainer: 'order-2 max-w-[70%]',
      },
      promoClasses: {
        promoContainer:
          'relative bg-theme-bg h-full flex flex-col ml:flex-row items-center justify-between p-m ml:px-xl ml:py-l before:content-[""] before:absolute before:w-screen before:h-full before:bg-theme-bg before:top-0 before:-z-10 ml:before:hidden',
        promoImageWrapper: 'w-full ml:w-3/5 mb-xxs ml:mb-0',
        promoContentWrapper: 'w-full ml:w-2/6',
        promoHeadline: 'text-s ml:text-m text-theme-text font-bold mb-s',
        promoSubheadline: 'text-sm-s ml:text-s text-theme-text font-medium mb-s',
        promoBody: 'text-body text-dark-gray mb-m',
        promoCTA: {
          wrapper: 'flex-col ml:items-center',
          cta1Classes: 'mb-0 ml:mr-2',
        },
      },
    },
  },
  rba: {
    classes: {
      headline: 'text-m ml:text-l font-medium',
      tabsClasses: {
        tabWrapper: 'flex flex-col ml:flex-row ml:min-h-[503px]',
        tabHeadlineWrapper: 'flex items-center pt-s mb-xs ml:mb-xxxs uppercase',
        tabHeadlineIcon: 'mr-xxxs',
        tabHeadline: '!font-serif text-xxs font-bold',
        tabTitlesContainer: 'relative flex order-3 ml:order-none pr-s pl-s',
        leftArrowIcon:
          'absolute right-full top-[40%] -translate-y-1/2 cursor-pointer bg-white before:absolute before:left-full before:inline-block before:h-[30px] before:w-[30px] before:bg-gradient-to-r before:from-white before:content-[""] ml:hidden',
        rightArrowIcon:
          'absolute left-full top-[40%] -translate-y-1/2 cursor-pointer bg-white before:absolute before:right-full before:inline-block before:h-[30px] before:w-[30px] before:bg-gradient-to-l before:from-white before:content-[""] ml:hidden',
        tabTitlesWrapper:
          'order-3 mb-xxs ml:order-none flex ml:inline-block overflow-auto snap-x ml:snap-none',
        tabTitlesList: 'w-full ml:w-1/5 ml:pr-s flex flex-col ml:justify-center',
        tabTitle: 'text-body text-dark-gray cursor-pointer mr-m ml:mr-0 my-xxxs snap-start',
        activeTabTitle:
          '!text-secondary font-heavy ml:pl-[26px] border-b-3 ml:border-b-0 border-primary relative before:hidden ml:before:inline-block before:absolute before:content-[""] before:w-[20px] before:h-[3px] before:bg-primary before:top-[50%] before:left-0',
        tabPanel: 'w-full',
        toggleContainer: 'order-2 mb-s',
      },
      promoClasses: {
        promoContainer:
          'relative bg-theme-bg h-full w-full flex flex-col ml:flex-row items-center justify-between p-[32px] before:content-[""] before:absolute before:w-screen before:h-full before:bg-theme-bg before:top-0 before:-z-10 ml:before:hidden',
        promoImageWrapper: 'relative ml:static w-full ml:w-1/2 ml:mr-[32px] mb-xxs',
        promoContentWrapper: 'w-full ml:w-1/2',
        promoEyebrow: 'text-xxs text-theme-text font-bold mb-xxxs',
        promoHeadline: 'text-m text-theme-text font-medium mb-xxxs',
        promoSubheadline: 'text-m text-theme-text font-extra-light mb-xxxs',
        promoBody: 'text-body text-theme-body mb-s',
        promoCTA: {
          wrapper: 'flex-col ml:items-center',
          cta1Classes: 'mr-2',
        },
        promoImageCaptionWrapper: 'ml:absolute top-full left-0 my-xxs',
        promoImageCaption: 'relative text-xxs ml:text-body [&_a]:!text-primary',
      },
    },
  },
};
