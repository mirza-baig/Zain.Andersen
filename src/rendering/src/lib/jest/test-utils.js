// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
