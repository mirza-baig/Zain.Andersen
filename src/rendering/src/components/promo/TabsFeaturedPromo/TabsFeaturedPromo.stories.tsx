// Global
import { Story, Meta } from '@storybook/react';
// Local
import TabsFeaturedPromo, { TabsFeaturedPromoProps } from './TabsFeaturedPromo';
import defaultData from './TabsFeaturedPromo.mock-data';

export default {
  title: 'Promo/TabsFeaturedPromo',
  component: TabsFeaturedPromo,
} as Meta;
const Template: Story<TabsFeaturedPromoProps> = (props) => <TabsFeaturedPromo {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
