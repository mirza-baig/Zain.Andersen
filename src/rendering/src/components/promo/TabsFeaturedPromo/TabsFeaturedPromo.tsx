// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { Field, Text, withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { TabsFeaturedPromoTheme } from './TabsFeaturedPromo.theme';
import { ComponentProps } from 'lib/types/component-props';
import { Tabs } from 'src/helpers/Tabs';
import { Headline } from 'src/helpers/Headline';
import React, { useEffect, useRef, useState } from 'react';
import Promo from './Promo.helper';
import { SvgIcon } from 'src/helpers/SvgIcon';
import classNames from 'classnames';
import { Item } from '@sitecore-jss/sitecore-jss-nextjs';
import ToggleSwitch from 'src/helpers/ToggleSwitch/ToggleSwitch';
import { getBreakpoint, useCurrentScreenType } from 'lib/utils/get-screen-type';
import { normalizeGuid } from 'lib/utils/string-utils';

type TabItemsProps = {
  fields: {
    children: Array<Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsFeaturedPromo.TabsFeaturedPromo>;
    headlineLevel: Item;
    headlineText: Field<string>;
  };
};

type TabCollectionsProps = {
  fields: {
    children: Array<
      Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsFeaturedPromo.TabsFeaturedPromo &
        TabItemsProps
    >;
  };
};

export type TabsFeaturedPromoProps =
  Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsFeaturedPromo.TabsFeaturedPromo &
    TabCollectionsProps &
    ComponentProps;

const TabsFeaturedPromo = (props: TabsFeaturedPromoProps) => {
  const { fields } = props;
  const { children } = fields || {};

  const { themeName, themeData } = useTheme(TabsFeaturedPromoTheme);
  const { currentScreenWidth } = useCurrentScreenType();

  const titlesScroller = useRef<HTMLDivElement>(null);

  //#region initial states and data of tabs

  // ToggleSwitch
  const [activeSwitchIndex, setActiveSwitchIndex] = useState(0);
  //Sets visible tab group
  const [visiblePanelIndex, setVisiblePanelIndex] = useState(0);
  //Sets visible tab panel
  const [currentIndex, setCurrentIndex] = useState(-1);
  // Sets if hash should be updated
  const [shouldUpdateURL, setShouldUpdateURL] = useState(false);

  //#endregion

  useEffect(() => {
    const selectedId = window.location.hash;

    if (selectedId || fields?.defaultActiveTab) {
      for (let collectionIndex = 0; collectionIndex < children?.length; collectionIndex++) {
        let isMatchFound = false;
        for (
          let itemIndex = 0;
          itemIndex < children[collectionIndex].fields.children?.length;
          itemIndex++
        ) {
          if (selectedId) {
            const item = children[collectionIndex].fields.children[itemIndex];
            const idToCompare = item.fields.contentId.value || `tab-${normalizeGuid(item.id)}`;

            if (selectedId.split('#')[1] === idToCompare) {
              setActiveSwitchIndex(collectionIndex);
              setVisiblePanelIndex(collectionIndex);
              setCurrentIndex(itemIndex);
              isMatchFound = true;
              break;
            }
          } else if (
            children[collectionIndex].fields.children[itemIndex].fields.contentId.value ===
            (
              fields.defaultActiveTab as Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsFeaturedPromo.TabsFeaturedPromo
            )?.fields.contentId.value
          ) {
            setActiveSwitchIndex(collectionIndex);
            setVisiblePanelIndex(collectionIndex);
            setCurrentIndex(itemIndex);
            isMatchFound = true;
            break;
          }
        }
        if (isMatchFound) {
          break;
        }
      }
    } else {
      setCurrentIndex(0);
    }
    // "children" and "fields.defaultActiveTab" are the props coming from layout service. We can ignore this warning.
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  // get the tab title appear into view if the tabs are in scrollable section
  useEffect(() => {
    const selectedItem = fields?.children?.[visiblePanelIndex]?.fields.children[currentIndex];

    const selectedItemId =
      selectedItem?.fields?.contentId.value || `tab-${normalizeGuid(selectedItem?.id)}`;

    if (shouldUpdateURL && currentIndex >= 0) {
      history.pushState(history.state, window.location.href, `#${selectedItemId}`);
      setShouldUpdateURL(false);
    }

    if (titlesScroller.current && currentScreenWidth <= getBreakpoint('ml')) {
      // We're using ref of the parent node of the titles in order to avoid bugs which can occur
      // if this component is authored multiple times in single page
      titlesScroller.current
        .querySelector(`[id='tab-${selectedItemId}']`)
        ?.scrollIntoView({ behavior: 'smooth', block: 'nearest', inline: 'center' });
    }
    // "fields.children" can be ignored as it is coming from layout service. We can ignore this warning.
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [visiblePanelIndex, currentIndex, shouldUpdateURL, currentScreenWidth]);

  const getSwitchTitles = (): string[] => {
    const switchTitles = ['', ''];

    children.forEach(
      (
        collection: Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsFeaturedPromo.TabsFeaturedPromo,
        index: number
      ) => {
        switchTitles[index] = (collection?.fields?.mobileToggleTitle as Field<string>).value;
      }
    );

    return switchTitles;
  };

  const selectTab = (itemIndex: number, collectionIndex: number) => {
    setShouldUpdateURL(true);
    setCurrentIndex(itemIndex);
    setVisiblePanelIndex(collectionIndex);
  };

  const renderToggleSwitch = () => {
    if (currentScreenWidth <= getBreakpoint('ml')) {
      return (
        <ToggleSwitch
          switchTitles={getSwitchTitles()}
          activeSwitchIndex={activeSwitchIndex}
          toggleActiveSwtich={() => {
            const _activeSwitchIndex = activeSwitchIndex === 0 ? 1 : 0;
            setVisiblePanelIndex(_activeSwitchIndex);
            setActiveSwitchIndex(_activeSwitchIndex);
            setCurrentIndex(0);
          }}
        />
      );
    }
    return null;
  };

  const renderTitles = (
    collection: Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsFeaturedPromo.TabsFeaturedPromo,
    collectionIndex: number
  ) => {
    if (
      (currentScreenWidth <= getBreakpoint('ml') && collectionIndex === visiblePanelIndex) ||
      currentScreenWidth > getBreakpoint('ml')
    ) {
      return (
        <>
          {collection?.fields?.headlineText?.value !== '' && (
            <div className={themeData.classes.tabsClasses.tabHeadlineWrapper}>
              {themeName === 'aw' && (
                <SvgIcon
                  icon="orange-triangle"
                  className={themeData.classes.tabsClasses.tabHeadlineIcon}
                />
              )}
              <Headline
                fields={collection?.fields}
                classes={themeData.classes.tabsClasses.tabHeadline}
              />
            </div>
          )}
          <div ref={titlesScroller} className={themeData.classes.tabsClasses.tabTitlesContainer}>
            {currentIndex > 0 && (
              <div
                className={themeData.classes.tabsClasses.leftArrowIcon}
                onClick={() => setCurrentIndex(currentIndex - 1)}
              >
                <SvgIcon icon="arrow-left" />
              </div>
            )}
            <ul className={themeData.classes.tabsClasses.tabTitlesWrapper} key={collectionIndex}>
              {collection?.fields.children.map(
                (
                  item: Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsFeaturedPromo.TabsFeaturedPromo,
                  itemIndex: number
                ) => {
                  return (
                    <li
                      key={itemIndex}
                      id={item.fields?.contentId.value || `tab-${normalizeGuid(item.id)}`}
                      className={classNames(
                        currentScreenWidth >= getBreakpoint('ml')
                          ? 'break-words'
                          : 'whitespace-nowrap',
                        themeData.classes.tabsClasses.tabTitle,
                        itemIndex === currentIndex &&
                          collectionIndex === visiblePanelIndex &&
                          themeData.classes.tabsClasses.activeTabTitle
                      )}
                      onClick={() => selectTab(itemIndex, collectionIndex)}
                      onKeyDown={(e) => {
                        if (e.code === 'Enter' || e.code === 'Space') {
                          selectTab(itemIndex, collectionIndex);
                        }
                      }}
                      tabIndex={0}
                    >
                      <Text tag="h4" field={item.fields?.tabTitle as Field<string>} />
                    </li>
                  );
                }
              )}
            </ul>
            {currentIndex < collection?.fields.children?.length - 1 && (
              <div
                className={themeData.classes.tabsClasses.rightArrowIcon}
                onClick={() => setCurrentIndex(currentIndex + 1)}
              >
                <SvgIcon icon="arrow-right" />
              </div>
            )}
          </div>
        </>
      );
    }

    return null;
  };

  if (!fields) {
    return null;
  }

  const renderTabTitles = () => {
    return (
      <>
        {/* Toggle Switch */}
        {children?.length > 1 && (
          <div className={themeData.classes.tabsClasses.toggleContainer}>
            {renderToggleSwitch()}
          </div>
        )}
        {/* Tab Titles List */}
        {children?.map(
          (
            collection: Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsFeaturedPromo.TabsFeaturedPromo,
            collectionIndex: number
          ) => {
            return (
              <React.Fragment key={collectionIndex}>
                {renderTitles(collection, collectionIndex)}
              </React.Fragment>
            );
          }
        )}
      </>
    );
  };

  const renderTabPanels = (): React.ReactElement[] => {
    const panels: Array<React.ReactElement> = [];
    children?.map(
      (
        collection: Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsFeaturedPromo.TabsFeaturedPromo,
        collectionIndex: number
      ) => {
        return (
          collectionIndex === visiblePanelIndex &&
          collection.fields?.children.forEach(
            (
              tabItem: Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsFeaturedPromo.TabsFeaturedPromo,
              tabIndex: number
            ) => {
              panels.push(
                <Promo key={tabIndex} classes={themeData.classes.promoClasses} {...tabItem} />
              );
            }
          )
        );
      }
    );

    return panels;
  };

  return (
    <Component variant="lg" dataComponent="promo/tabsfeaturedpromo" {...props}>
      <div className="col-span-12">
        <Headline fields={fields} classes={themeData.classes.headline} />
      </div>
      {fields.children?.length > 0 && (
        <div className="col-span-12 mb-ml">
          <Tabs classes={themeData.classes.tabsClasses} currentTabIndex={currentIndex}>
            {renderTabTitles()}
            {renderTabPanels()}
          </Tabs>
        </div>
      )}
    </Component>
  );
};

export default withDatasourceCheck()<TabsFeaturedPromoProps>(TabsFeaturedPromo);
