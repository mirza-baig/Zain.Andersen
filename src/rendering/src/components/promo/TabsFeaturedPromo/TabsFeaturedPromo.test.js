// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import TabsFeaturedPromo from './TabsFeaturedPromo';
import defaultData from './TabsFeaturedPromo.mock-data';

it('renders correctly', async () => {
  const component = snapshot(TabsFeaturedPromo, { componentProps: defaultData });
  await hasDataComponent(component, 'promo/tabsfeaturedpromo');
});
