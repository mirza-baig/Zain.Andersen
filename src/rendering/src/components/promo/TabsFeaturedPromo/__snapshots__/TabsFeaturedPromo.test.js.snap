// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="promo/tabsfeaturedpromo"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="col-span-12"
      >
        <div
          class="items-top font-sans [&_a:hover]:underline text-s ml:text-m font-bold"
        >
          <h2>
            The right solution for any project
          </h2>
        </div>
      </div>
      <div
        class="col-span-12 mb-ml"
      >
        <div
          class="flex flex-col ml:flex-row min-h-[542px]"
        >
          <div
            class="w-full ml:w-1/4 ml:pr-s flex flex-col ml:inline-block"
          >
            <div
              class="order-2 max-w-[70%]"
            />
            <div
              class="flex items-center border-t pt-s mb-xxs uppercase"
            >
              <span
                class="mr-xxxs"
              />
              <div
                class="items-top font-sans [&_a:hover]:underline text-xs font-heavy"
              >
                <h2>
                  Doors by Type
                </h2>
              </div>
            </div>
            <div
              class="relative flex order-3 ml:order-none pr-s pl-s"
            >
              <div
                class="absolute right-full top-1/2 -translate-y-1/2 cursor-pointer bg-white before:absolute before:left-full before:inline-block before:h-[30px] before:w-[30px] before:bg-gradient-to-r before:from-white before:content-[""] ml:hidden"
              >
                <span />
              </div>
              <ul
                class="order-3 flex ml:inline-block overflow-auto snap-x ml:snap-none"
              >
                <li
                  class="break-words text-body ml:text-xs text-dark-gray cursor-pointer mx-xxs ml:mx-0 py-xxxs ml:px-xxs ml:border-l-2 my-s snap-start"
                  id="awning"
                  tabindex="0"
                >
                  <h4>
                    Awning windows
                  </h4>
                </li>
                <li
                  class="break-words text-body ml:text-xs text-dark-gray cursor-pointer mx-xxs ml:mx-0 py-xxxs ml:px-xxs ml:border-l-2 my-s snap-start !text-secondary font-heavy border-b-2 ml:border-b-0 border-primary"
                  id="bay-bow"
                  tabindex="0"
                >
                  <h4>
                    Bay & bow windows
                  </h4>
                </li>
                <li
                  class="break-words text-body ml:text-xs text-dark-gray cursor-pointer mx-xxs ml:mx-0 py-xxxs ml:px-xxs ml:border-l-2 my-s snap-start"
                  id="big-doors"
                  tabindex="0"
                >
                  <h4>
                    Big doors
                  </h4>
                </li>
              </ul>
              <div
                class="absolute left-full top-1/2 -translate-y-1/2 cursor-pointer bg-white before:absolute before:right-full before:inline-block before:h-[30px] before:w-[30px] before:bg-gradient-to-l before:from-white before:content-[""] ml:hidden"
              >
                <span />
              </div>
            </div>
            <div
              class="flex items-center border-t pt-s mb-xxs uppercase"
            >
              <span
                class="mr-xxxs"
              />
              <div
                class="items-top font-sans [&_a:hover]:underline text-xs font-heavy"
              >
                <h2>
                  Test Tab section
                </h2>
              </div>
            </div>
            <div
              class="relative flex order-3 ml:order-none pr-s pl-s"
            >
              <div
                class="absolute right-full top-1/2 -translate-y-1/2 cursor-pointer bg-white before:absolute before:left-full before:inline-block before:h-[30px] before:w-[30px] before:bg-gradient-to-r before:from-white before:content-[""] ml:hidden"
              >
                <span />
              </div>
              <ul
                class="order-3 flex ml:inline-block overflow-auto snap-x ml:snap-none"
              />
            </div>
            <div
              class="flex items-center border-t pt-s mb-xxs uppercase"
            >
              <span
                class="mr-xxxs"
              />
              <div
                class="items-top font-sans [&_a:hover]:underline text-xs font-heavy"
              >
                <h2>
                  Types of Windows
                </h2>
              </div>
            </div>
            <div
              class="relative flex order-3 ml:order-none pr-s pl-s"
            >
              <div
                class="absolute right-full top-1/2 -translate-y-1/2 cursor-pointer bg-white before:absolute before:left-full before:inline-block before:h-[30px] before:w-[30px] before:bg-gradient-to-r before:from-white before:content-[""] ml:hidden"
              >
                <span />
              </div>
              <ul
                class="order-3 flex ml:inline-block overflow-auto snap-x ml:snap-none"
              >
                <li
                  class="break-words text-body ml:text-xs text-dark-gray cursor-pointer mx-xxs ml:mx-0 py-xxxs ml:px-xxs ml:border-l-2 my-s snap-start"
                  id="casement"
                  tabindex="0"
                >
                  <h4>
                    Casement windows
                  </h4>
                </li>
                <li
                  class="break-words text-body ml:text-xs text-dark-gray cursor-pointer mx-xxs ml:mx-0 py-xxxs ml:px-xxs ml:border-l-2 my-s snap-start"
                  id="entry"
                  tabindex="0"
                >
                  <h4>
                    Entry doors
                  </h4>
                </li>
                <li
                  class="break-words text-body ml:text-xs text-dark-gray cursor-pointer mx-xxs ml:mx-0 py-xxxs ml:px-xxs ml:border-l-2 my-s snap-start"
                  id="small-doors"
                  tabindex="0"
                >
                  <h4>
                    Small doors
                  </h4>
                </li>
              </ul>
              <div
                class="absolute left-full top-1/2 -translate-y-1/2 cursor-pointer bg-white before:absolute before:right-full before:inline-block before:h-[30px] before:w-[30px] before:bg-gradient-to-l before:from-white before:content-[""] ml:hidden"
              >
                <span />
              </div>
            </div>
          </div>
          <div
            class="w-full ml:w-3/4"
          >
            <div
              class="theme-gray relative bg-theme-bg h-full flex flex-col ml:flex-row items-center justify-between p-m ml:px-xl ml:py-l before:content-[""] before:absolute before:w-screen before:h-full before:bg-theme-bg before:top-0 before:-z-10 ml:before:hidden"
            >
              <div
                class="w-full ml:w-3/5 mb-xxs ml:mb-0"
              >
                <div
                  class="relative hidden md:block aspect-picture   "
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 133.33333333333331% 0px 0px 0px;"
                    />
                    <img
                      alt=""
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
                <div
                  class="relative w-full md:hidden"
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 133.33333333333331% 0px 0px 0px;"
                    />
                    <img
                      alt=""
                      data-nimg="responsive"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                    />
                    <noscript />
                  </span>
                </div>
              </div>
              <div
                class="w-full ml:w-2/6"
              >
                <div
                  class="items-top font-sans [&_a:hover]:underline text-s ml:text-m text-theme-text font-bold mb-s"
                >
                  <h2>
                    Bay & bow windows
                  </h2>
                </div>
                <div
                  class="text-body text-dark-gray mb-m"
                >
                  <div
                    class="body-copy"
                  >
                    <div>
                      <strong>
                        What: 
                      </strong>
                      The cozy nook with at least three windows
                      <br />
                      

                      <strong>
                        Where:
                      </strong>
                       Most often used in living rooms and bedrooms
                      <br />
                      

                      <strong>
                        Fun Fact:
                      </strong>
                       Angled out beyond the house wall, for extra space inside
                    </div>
                  </div>
                </div>
                <div
                  class="mb-s flex items-start md:flex-row md:items-center"
                >
                  <a
                    aria-label="Learn More"
                    class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
                    data-gtm-click=""
                    data-gtm-dl-event="cta_click"
                    href="/"
                    role="link"
                    tabindex="0"
                    target=""
                    title=""
                  >
                    Learn More
                    <span
                      class="ml-xxs"
                    />
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
